### ISSUE TITLE
Prevent users from scrubbing outside of live buffered range

### ISSUE THREAD

**Issue by baconz** [google/shaka-player#425]
We would like to prevent users from scrubbing past the buffered range for live content. If a user scrubs all the way to the live edge and `bufferedAhead` falls below 1, the player triggers a buffering event, and since new segments aren't available yet, the user has to wait for the entire `rebufferingGoal` to resume watching.

We have a similar problem on startup. If the playhead starts on the live edge, the user has to wait for the entire `rebufferingGoal` to start playback.

Our workaround for this had been to add `minBufferTime` to our AST so that the user can't scrub too close to live, and so that they can fill up the buffer on start-up. The problem with this approach is that the player ignores segments that are past the `availabilityEnd`, and so our "extra" segments don't get buffered.

Would it be crazy to add a flag that tells shaka to aggressively load segments even if they are outside the availability range? I know this is a bit out of spec, but it would be super useful here!

Alternatively, could we change the delay logic `getSeekRangeEnd()` to something like:

```
  // I know we don't have minBufferTime in presentation_timeline right now, but...
  var distanceFromEnd = this.duration_ - this.getSegmentAvailabilityEnd();

  var delay;
  if (this.duration_ == Number.POSITIVE_INFINITY ||  distanceFromEnd > this.minBufferTime_ ) {
    // Normal live playback or an IPR that is not at its end, don't let the user scrub past the buffer
    delay = this.presentationDelay_ + this.minBufferTime_;
  } else {
    // This is VOD or the end of an in progress recording.
    // delay will always be 0 for VOD, and it will 
    delay = distanceFromEnd;
  }
```

**Comment by TheModMaker** [comment]
Does your manifest have a `suggestedPresentationDelay`?  This will delay the seek range from the live edge and the Player will still download segments past it.  So if you set `suggestedPresentationDelay` to 5 seconds, seeking to the end of the seek range will still have 5 seconds of buffer ahead of it.

**Comment by baconz** [comment]
Ah, I missed 7aab332e9bea92cd5bf811e558ab531b2c3b190f.

It seems like there would still be an issue for in-progress recordings, though. We would want the user to be able to scrub to the "end," past the delay, but only after it becomes available. So a slight modification of my code above:

```
  var distanceFromEnd = this.duration_ - this.getSegmentAvailabilityEnd();

  var delay;
  if (this.duration_ == Number.POSITIVE_INFINITY ||  distanceFromEnd > this.minBufferTime_ ) {
    // Normal live playback or an IPR that is not at its end, don't let the user scrub past the buffer
    delay = this.presentationDelay_
  } else {
    // This is VOD or the end of an in progress recording.
    // delay will always be 0 for VOD, and it will 
    delay = distanceFromEnd;
  }
```

**Comment by baconz** [comment]
Wanted to bump this up. I'm happy to put it into a PR if that is helpful.

To sum up: Now that `suggestedPresentationDelay` is being applied correctly as a delay to `seekEnd()` we will never allow users to scrub to the end of in progress recordings.

**Comment by tdrews** [comment]
Yes, a PR would be great. Thanks.

**Comment by joeyparrish** [comment]
Looks like this was addressed by PR #441, which was merged.  Closing.
